.travis_before_install.bash

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+if [ "$TRAVIS_OS_NAME" == "linux" ]
+then
+    if [ "$CXX" == "g++" ]
```

Comment:
Both clang and GCC builds need the newer libstdc++ to have filesystem support so GCC 9 should be installed unconditionally 
```suggestion
    sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
    if [ "$CXX" == "clang++" ]
    then
        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
        sudo apt-add-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-8 main"
    fi
```